src/client/datascience/kernel-launcher/kernelLauncher.ts

```diff
@@ -69,9 +73,10 @@ class KernelProcess implements IKernelProcess {
         args[indexOfConnectionFile] = this.connectionFile.filePath;
         args.splice(0, 1);
 
-        const executionService = await this.executionFactory.create({
+        const executionService = await this.executionFactory.createActivatedEnvironment({
             resource: undefined,
-            pythonPath: this._kernelSpec.path
+            interpreter: matchingInterpreter,
+            extraVars: this._kernelSpec.env
         });
         const exeObs = executionService.execObservable(args, {});
```

Comment:
```suggestion
        const exeObs = executionService.execObservable(args, {env: this._kernelSpec.env});
```

This should be sufficient.